var/spack/repos/builtin/packages/discotec/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Discotec(CMakePackage):
+    """This project contains DisCoTec, a code for the distributed sparse
+    grid combination technique with MPI parallelization."""
+
+    homepage = "https://github.com/SGpp/DisCoTec"
+    url = "https://github.com/SGpp/DisCoTec"
+    git = "https://github.com/SGpp/DisCoTec"
+
+    # notify when the package is updated.
+    maintainers = ["freifrauvonbleifrei", "pfluegdk"]
+
+    version("main", branch="main")
+
+    variant("ft", default=False, description="DisCoTec with algorithm-based fault tolerance")
+    variant("gene", default=False, description="Build for GENE (as task library)")
+    variant("hdf5", default=True, description="Interpolation output with HDF5")
+    variant("lto", default=True, description="Build with link-time optimization")
+    variant("openmp", default=True, description="Parallelize with OpenMP")
+    variant("timing", default=True, description="With high-res timers")
+    variant("test", default=True, description="Build Boost tests")
+    variant("selalib", default=False, description="Build selalib example")
+    variant("vtk", default=False, description="Build with VTK support")
+
+    depends_on(
+        "boost +test +serialization +filesystem +system +program_options", type=("build", "run")
+    )
+    depends_on("cmake@3.24.2:", type="build")
+    depends_on("glpk")
+    depends_on("highfive+mpi+boost+ipo", when="+hdf5")
+    depends_on("mpi", type=("build", "run"))
```

Comment:
Why `mpi` is of type `build,run`? Don't you `link` to it?

Reply:
yes, but the mpi compilers need to be used for building, and `mpirun` (or similar) needs to be used for executing (and the same mpi versions at that). is `link` something I should add to the list?

Reply:
I would use the default and remove `type=` alltogether. That's what most `mpi` software do.